substrate/frame/support/src/generate_genesis_config.rs

```diff
@@ -211,86 +220,160 @@ macro_rules! build_struct_json_patch {
 		$($struct_type:ident)::+ { $($tail:tt)* }
 	) => {
 		{
-			let mut keys = $crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
+			let mut __keys = $crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
 			#[allow(clippy::needless_update)]
-			let struct_instance = $crate::build_struct_json_patch!($($struct_type)::+, keys @  { $($tail)* });
-			let mut json_value =
-				$crate::__private::serde_json::to_value(struct_instance).expect("serialization to json should work. qed");
-			$crate::generate_genesis_config::retain_initialized_fields(&mut json_value, &keys, Default::default());
-			json_value
+			let __struct_instance = $crate::build_struct_json_patch!($($struct_type)::+, __keys @  { $($tail)* }).0;
+			let mut __json_value =
+				$crate::__private::serde_json::to_value(__struct_instance).expect("serialization to json should work. qed");
+			$crate::generate_genesis_config::retain_initialized_fields(&mut __json_value, &__keys, Default::default());
+			__json_value
 		}
 	};
 	($($struct_type:ident)::+, $all_keys:ident @ { $($tail:tt)* }) => {
-		$($struct_type)::+ {
-			..$crate::build_struct_json_patch!($($struct_type)::+, $all_keys @ $($tail)*)
+		{
+			let __value = $crate::build_struct_json_patch!($($struct_type)::+, $all_keys @ $($tail)*);
+			(
+				$($struct_type)::+ { ..__value.0 },
+				__value.1
+			)
 		}
 	};
 	($($struct_type:ident)::+, $all_keys:ident  @  $key:ident: $($type:ident)::+ { $keyi:ident : $value:tt }  ) => {
-		$($struct_type)::+ {
+		(
+			$($struct_type)::+ {
 			$key: {
 				$all_keys.push($crate::generate_genesis_config::InitializedField::partial(stringify!($key)));
 				$all_keys.push(
-					$crate::generate_genesis_config::InitializedField::full(concat!(stringify!($key), ".", stringify!($keyi)))
+						$crate::generate_genesis_config::InitializedField::full(
+							concat!(stringify!($key), ".", stringify!($keyi))
+						)
 				);
 				$($type)::+ {
 					$keyi:$value,
 					..Default::default()
 				}
 			},
 			..Default::default()
-		}
+			},
+			$crate::generate_genesis_config::InitilizationType::Partial
+		)
 	};
```

Comment:
what is this branch exactly?

It get triggered only when value is a single token without comma, and key is the last key, no?

so triggered for:
```rust
build_struct_json_patch! ( RuntimeGenesisConfig {
	a_field: ConfigA {
		inner_field: 32
	}
}),
```

but not for:
```rust
build_struct_json_patch! ( RuntimeGenesisConfig {
	a_field: ConfigA {
		inner_field: 32,
	}
}),
```

nor for:
```rust
build_struct_json_patch! ( RuntimeGenesisConfig {
	a_field: ConfigA {
		inner_field: MyEnum::Bar
	}
}),
```

It doesn't look necessary.

Also I removed it and all tests pass.

Reply:
Good catch, indeed it seems to be unused! It is a left-over from some of my previous iterations. Thanks for playing with this!

Reply:
removed in: b830709